pcdsdevices/eps.py

```diff
@@ -0,0 +1,68 @@
+"""
+Classes for Equipment Protection System (EPS) readbacks.
+
+This covers generalized ways that we can protect hardware from
+damaging itself. For example: we can prevent a motor from moving
+forward if moving forward would collide with another motor.
+"""
+from ophyd import Component as Cpt
+from ophyd import Device
+
+from .interface import BaseInterface
+from .signal import InternalSignal, PytmcSignal
+from .variety import set_metadata
+
+
+class EPS(BaseInterface, Device):
+    """
+    Corresponds to DUT_EPS from lcls-twincat-general.
+
+    This struct represents a combined EPS status that can be
+    made up of many independent protection factors. It summarizes
+    all of the EPS information in a way that can be consumed
+    by higher level applications so that the user knows exactly
+    why their motor cannot move, why their valve cannot open,
+    etc.
+
+    This has a corresponding EPS widget in pcdswidgets, so
+    this class only makes minimal adjustments to make the default
+    typhos view usable.
+    """
+    eps_ok = Cpt(PytmcSignal, "bEPS_OK", io="i", kind="normal",
+                 doc="EPS summary: true if everything is OK")
+    message = Cpt(PytmcSignal, "sMessage", io="i", kind="normal",
+                  string=True,
+                  doc="Message from EPS to the user.")
+    flags_raw = Cpt(PytmcSignal, "nFlags", io="i", kind="omitted",
+                    doc="Raw EPS bitmask")
+    flags = Cpt(InternalSignal, kind="normal")
+    flag_desc = Cpt(PytmcSignal, "sFlagDesc", io="i", kind="omitted",
+                    string=True,
+                    doc="Semicolon-delimited descriptions of each flag")
+
+    # See _update_flag docstring
+    set_metadata(flags, dict(variety="bitmask", bits=8))
+
+    @flags_raw.sub_value
+    def _update_flag(self, value, *args, **kwargs):
+        """
+        Update the flags value to a form that is usable in a pydm bitmask.
+
+        If bit 32 is TRUE, this comes as a negative number because EPICS CA has
```

Comment:
Not much to add on this other than - great description that'll assuredly help us once this falls out of working memory 👍